### ISSUE TITLE
Internal Server Error when donning a set of Aquarium Armor

### ISSUE THREAD

**Issue by zderApfel** [MinecraftModDevelopmentMods/BaseMetals#470]
Client gets internal server error, console here: https://pastebin.com/heUWCXfN

Base Metals version: rc2.332 for Forge 1.12.2

**Comment by jriwanek** [comment]
Please provide full logs, not just the parts you consider relevent.

**Comment by zderApfel** [comment]
You want me to post my full server console? I gave you all the errors that came up when the user is kicked 


Sent from Yahoo Mail for iPhone


On Thursday, July 30, 2020, 4:09 PM, Jasmine Iwanek <notifications@github.com> wrote:




Please provide full logs, not just the parts you consider relevent.

**Comment by jriwanek** [comment]
I need the full logs, including server start ideally. also, the version of MMDLib you're using.

**Comment by dshadowwolf** [comment]
hrm, actually... This looks like I might have dropped the ball and have a client-code reference in a shared section.

```
30.07 17:34:53 [Server] INFO Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/Minecraft
30.07 17:34:53 [Server] INFO at com.mcmoddev.basemetals.properties.AquariumProperty.apply(AquariumProperty.java:33) ~[AquariumProperty.class:?]
```

Though the error line doesn't map out properly, this looks like where we do a "World Lookup", that is...
```java
				World w = com.mcmoddev.basemetals.BaseMetals.proxy.getWorld(player.dimension);
				if(w == null) return;
```

If that code doesn't work just right - or if -rc2.232 went out before I fixed it to have a proxy-based lookup - then that could be the trigger and the -dev build would have this resolved already.

**Comment by zderApfel** [comment]
Understood.

I'm using MMDLib-1.12-1.3.3.55

Unfortunately the logs from earlier are already gone so I went ahead and replicated the error here: https://pastebin.com/bTQHNh0y

**Comment by dshadowwolf** [comment]
Okay, this problem has already been fixed - it was noted as existing in -rc2.332 and is fixed in -dev but the fix has not, yet, been released as we've been somewhat overwhelmed by IRL issues or other work. Want to push a fix out, still under the -rc2 banner, @jriwanek  ?

**Comment by jriwanek** [comment]
I will soonish.

**Comment by zderApfel** [comment]
> I will soonish.

Take your time, I've just advised my server to not use the armor until a fix comes out.

**Comment by jamesiscool33** [comment]
Please fix this @jriwanek my friend group server has justg had someone kicked and he cant rejoin :( also @zderApfel have you found a way around this like clearing their inventory?

**Comment by zderApfel** [comment]
> Please fix this @jriwanek my friend group server has justg had someone kicked and he cant rejoin :( also @zderApfel have you found a way around this like clearing their inventory?

Yes I did. What you have to do is pull their player data file from the server and then open it in a program like NBT Explorer. Then you have to search through their inventory until you find the armor and delete it.

Also I'd just like to declare the server we were playing on has been long gone for a while, but I never got confirmation that the fix was published
